sftp.go

```diff
@@ -227,3 +231,25 @@ func (s *StatusError) Error() string {
 func (s *StatusError) FxCode() fxerr {
 	return fxerr(s.Code)
 }
+
+func isSFTPExtensionSupported(extension SSHExtensionPair) bool {
+	for _, supportedExtension := range supportedSFTPExtensions {
```

Comment:
Why not a `map` rather than having to iterate through the entire list of supported extensions?

Reply:
well `sftpExtensions` must be a slice, so I created `supportedSFTPExtensions` as a slice too so we can easily define:

```
sftpExtensions = supportedSFTPExtensions
```

We can define a map too but this way we duplicate the values, or we can create a map only and then iterate the map to populate `sftpExtensions`. I have no preference, please let me know how I have to modify the patch, thanks!

Am I missing something?

Reply:
IMO keeping it a slice right now is fine as that is required for the current packet setup. It does mean having to loop over , but it is and probably always will be a very short list so I don't think converting it will make much difference.

If it gets annoying we can change the sshFxVersionPacket struct to hold a map instead of a slice and convert things over.